blob/s3blob/s3blob.go

```diff
@@ -366,6 +377,89 @@ func (b *bucket) ListPaged(ctx context.Context, opts *driver.ListOptions) (*driv
 	return &page, nil
 }
 
+func (b *bucket) legacyListPaged(ctx context.Context, opts *driver.ListOptions) (*driver.ListPage, error) {
+	pageSize := opts.PageSize
```

Comment:
It looks like a high percentage of the logic is shared between `ListPaged` and `legacyListPaged`. Could you try sharing more of it? For example, you could make only the `ListObjectsRequest` call conditional, and in the legacy version convert from the V2 objects to the V1 objects (and from V1 to V2 in the result).

Reply:
Hi @vangent,

I've updated the code (see 24d73e47894c24b023fc082c5bdc8290d3c4919b). To me the previous solution looks cleaner. What do you think?

Reply:
I like this one, thanks!